diff --git a/networking/tc.c b/networking/tc.c
--- a/networking/tc.c
+++ b/networking/tc.c
@@ -233,6 +233,10 @@
 #endif
 static int cbq_print_opt(struct rtattr *opt)
 {
+#if !defined(TCA_CBQ_MAX) || !defined(TCA_CBQ_RATE) || !defined(TC_CBQ_MAXPRIO)
+	(void)opt;
+	return 0;
+#else
 	struct rtattr *tb[TCA_CBQ_MAX+1];
 	struct tc_ratespec *r = NULL;
 	struct tc_cbq_lssopt *lss = NULL;
@@ -321,6 +325,7 @@
 	}
  done:
 	return 0;
+#endif
 }
 
 static FAST_FUNC int print_qdisc(
